### ISSUE TITLE
Wrong metadata data types for PostgreSQL DB

### ISSUE THREAD

**Issue by tztztztz** [phalcon/cphalcon#12842]
Method \Phalcon\Mvc\Model\MetaData::getDataTypes is returning wrong datatype for PostgreSQL database and column types double precision and real. 

It is returning 2 which wrongly corresponding to \Phalcon\Db\Column ::TYPE_VARCHAR instead of respectively: \Phalcon\Db\Column::TYPE_DOUBLE for double precision and \Phalcon\Db\Column::TYPE_FLOAT for real.

Example dump of print_r of array returned by getDataTypes($model), where $model is subclass of \Phalcon\Mvc\Model:

```
$dataTypes = $entity->getModelsMetaData()->getDataTypes($entity);
echo '<pre>'; print_r($dataTypes); exit;
Array
(
    [id] => 0
    [name] => 2
    [is_self_delivery] => 8
    [is_active_for_client] => 8
    [is_active_for_admin] => 8
    [default_delivery_price_netto] => 2
    [is_default] => 8
    [can_delivery_to_client] => 8
    [can_delivery_returns] => 8
    [php_manager_class] => 2
    [default_delivery_price_vat_pct] => 2
    [default_delivery_price_vat_val] => 2
    [default_delivery_price_brutto] => 2
)
```

Table structure dump from Postgresql version 9.5:

```
CREATE TABLE public.k3_delivery_method
(
  id integer NOT NULL DEFAULT nextval('k3_delivery_method_id_seq'::regclass),
  name character varying(128),
  is_self_delivery boolean NOT NULL DEFAULT false,
  is_active_for_client boolean NOT NULL DEFAULT false,
  is_active_for_admin boolean NOT NULL DEFAULT false,
  default_delivery_price_netto double precision NOT NULL,
  is_default boolean NOT NULL DEFAULT false,
  can_delivery_to_client boolean NOT NULL DEFAULT true,
  can_delivery_returns boolean NOT NULL DEFAULT false,
  php_manager_class character varying(255),
  default_delivery_price_vat_pct real NOT NULL DEFAULT 0,
  default_delivery_price_vat_val double precision NOT NULL DEFAULT 0,
  default_delivery_price_brutto double precision NOT NULL DEFAULT 0,
  CONSTRAINT k3_delivery_method_pkey PRIMARY KEY (id)
)
WITH (
  OIDS=FALSE
);
ALTER TABLE public.k3_delivery_method
  OWNER TO postgres;
```

System details:

PHALCON:
Web framework delivered as a C-extension for PHP
phalcon => enabled
Author => Phalcon Team and contributors
Version => 3.1.2
Build Date => Apr  5 2017 15:48:35
Powered by Zephir => Version 0.9.7-1fae5e50ac

Directive => Local Value => Master Value
phalcon.db.escape_identifiers => On => On
phalcon.db.force_casting => Off => Off
phalcon.orm.events => On => On
phalcon.orm.virtual_foreign_keys => On => On
phalcon.orm.column_renaming => On => On
phalcon.orm.not_null_validations => On => On
phalcon.orm.exception_on_failed_save => Off => Off
phalcon.orm.enable_literals => On => On
phalcon.orm.late_state_binding => Off => Off
phalcon.orm.enable_implicit_joins => On => On
phalcon.orm.cast_on_hydrate => Off => Off
phalcon.orm.ignore_unknown_columns => Off => Off

PHP:
PHP 5.6.21 (cli) (built: Apr 27 2016 20:13:58) 
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies 

System:
Windows 10

Webserver:
Nginx

**Comment by tztztztz** [comment]
Is anyone up to fix this problem? It should be easy to do so....

**Comment by Jurigag** [comment]
You can always create PR if you want.

**Comment by tztztztz** [comment]
Whats PR?

**Comment by Jurigag** [comment]
Pull Request

**Comment by tztztztz** [comment]
I dont know hpow to do pull request and i cant use Zephyr on windows.

Can you fix it, this is like 5 min job for someone who have Zephyr ready.

**Comment by Jurigag** [comment]
You don't need zephir for it. You never used git/github? Oh okay.

**Comment by tztztztz** [comment]
No, i have account at github because it's required by Phalcon Phorum.

I doesn't understan why i dont need Zephyr - source code of Phalcon is written in Zephyr, so if i wanted to fix it i need to use Zephyr, right?

**Comment by Jurigag** [comment]
Yes but you can use any editor to edit zephir code, you don't need to install it to fix anything.

**Comment by tztztztz** [comment]
Ok, but how can i test it? You want me to write code without compiling and testing it?

**Comment by Jurigag** [comment]
By making PR, it will compile and test in travis.

**Comment by tztztztz** [comment]
I can't do it, i don't know github, it would take probably several hours for me before i woud find out how to create pull request. Why you can't fix it?

**Comment by tztztztz** [comment]
but how?

**Comment by Jurigag** [comment]
I can fix it, just don't have time right now for it. What you mean how? Just by making PR.

1. Fork repo
2. Edit code in repository
3. https://help.github.com/articles/creating-a-pull-request/

Pretty simple and easy really.

**Comment by tztztztz** [comment]
Sorry but im unable to do it. Ok then, bug stays.

**Comment by tztztztz** [comment]
Is this file responsible for mapping or im wrong:

https://github.com/tztztztz/cphalcon/blob/master/phalcon/mvc/model/metadata/strategy/annotations.zep

??

**Comment by tztztztz** [comment]
Come on at least tell me what file i must change.

**Comment by tztztztz** [comment]
Is there any way to search entire forked code?

**Comment by Jurigag** [comment]
https://github.com/tztztztz/cphalcon/blob/master/phalcon/db/adapter/pdo/postgresql.zep this file most likely describeColumns method

**Comment by tztztztz** [comment]
Thanks

**Comment by tztztztz** [comment]
Ok let's see what happens now :|

**Comment by Jurigag** [comment]
You need to use 3.2.x branch and also update changelog.md

**Comment by tztztztz** [comment]
I dont know how to do it

**Comment by tztztztz** [comment]
but i changed the code, can you pull request for me copying code?

**Comment by tztztztz** [comment]
Just copy my work and pull request ok?

**Comment by tztztztz** [comment]
Is it ok now... I lost so much time learning it, just to make small change :/

**Comment by sergeyklay** [comment]
Fixed in the `3.2.x branch`. Feel free to open a new issue if the problem appears again. Thank you for contributing.
